File: Source/TS/entities/entity.ts
import { Vector2 } from "../utilities/Vector2";

export abstract class Entity
{
    // Settings
    protected position: Vector2;
    protected velocity: Vector2;

    // Stats
    protected health: number = 100;
    protected melee: number = 0;
    protected mogging: number = 0;
    
    // Checks
    protected alive: boolean = true;

    constructor(position: Vector2)
    {
        this.position = position;
        this.velocity = Vector2.zero();
    }

    // Remove deltaTime if not planned, this is mainly just a placeholder
    abstract update(deltaTime: number): void;

    protected move(deltaTime: number): void
    {
        this.position.sum(this.velocity.clone().scale(deltaTime));
    }

    damage(amount: number): void
    {
        this.health -= amount;
        if (this.health <= 0)
        {
            this.die();
        }
    }

    protected die(): void
    {
        this.alive = false;
    }

    get getPosition()
    {
        return this.position.clone();
    }
}

File: core/src/main/java/io/github/RimIsland/Entities/Entity.java
package io.github.RimIsland.Entities;

import com.badlogic.gdx.math.Vector2;

public abstract class Entity
{
    // Settings
    protected Vector2 position;
    protected Vector2 velocity;

    // Stats
    protected int health = 100;
    protected int melee = 0;
    protected int mogging = 0;

    // Checks
    protected boolean alive = true;

    protected Entity(Vector2 position)
    {
        this.position = position;
        this.velocity = new Vector2().set(0,0);
    }

    abstract void update(int deltaTime);

    protected void move(int deltaTime)
    {
        this.position.add(this.velocity);
    }

    public void damage(int amount)
    {
        this.health -= amount;
        if (this.health <= 0)
        {
            this.die();
        }
    }

    protected void die()
    {
        this.alive = false;
    }

    public void setPosition(Vector2 position)
    {
        this.position = position;
    }

    public Vector2 getPosition()
    {
        return this.position;
    }

    public void setVelocity(Vector2 velocity)
    {
        this.velocity = velocity;
    }

    public Vector2 getVelocity()
    {
        return this.velocity;
    }
}
